include/nbl/system/IThreadHandler.h

```diff
@@ -0,0 +1,103 @@
+#ifndef __NBL_I_THREAD_HANDLER_H_INCLUDED__
+#define __NBL_I_THREAD_HANDLER_H_INCLUDED__
+
+#include <mutex>
+#include <condition_variable>
+#include <thread>
+
+namespace nbl {
+namespace system
+{
+
+// Usage:
+/*
+* class MyThreadHandler : public IThreadHandler<SomeInternalStateType> { .... };
+* 
+* MyThreadHandler handler;
+* std::thread thread(&MyThreadHandler::thread, &handler);
+* //...
+* //... communicate with the thread using your methods (see note at the end of this section)), thread will sleep until wakeupPredicate() returns true
+* //...
+* handler.terminate(thread);
+* After this handler can be safely destroyed.
```

Comment:
from your usage examples it appears that an object derived from `IThreadHandler` always comes in a 1:1 pair with an `std::thread`

Since your thread handler doesn't support multiple workers waiting on one condition, you could fold the `std::thread` into `IThreadHandler` as a member variable, start it in the constructor and fold `terminate` into the destructor

Reply:
at least observing all 3 usages in `COpenGL_LogicalDevice`,`COpenGL_Queue` and `COpenGL_Swapchain`

Reply:
i dont remember what was my reason for having thread separately from handler, now i dont see why wouldnt it be possible